src/docfx/build/Build.cs

```diff
@@ -36,33 +37,23 @@ private static async Task BuildDocset(string docsetPath, string outputPath, Comm
                 try
                 {
                     // load and trace entry repository
-                    var repositoryProvider = new RepositoryProvider(docsetPath, options);
+                    var repositoryProvider = new RepositoryProvider(docsetPath, options, () => restoreGitMap, () => config);
                     var repository = repositoryProvider.GetRepository(FileOrigin.Default);
                     Telemetry.SetRepository(repository?.Remote, repository?.Branch);
                     var locale = LocalizationUtility.GetLocale(repository, options);
 
-                    using (var restoreGitMap = RestoreGitMap.Create(docsetPath, locale))
+                    using (restoreGitMap = RestoreGitMap.Create(docsetPath, locale))
```

Comment:
if we can use fulll path of docset path as the restore file path hash, we can avoid the locale, then we can break the cycle.

Reply:
thanks for the hints, will try these ideas out in later PR